### ISSUE TITLE
lintr does not check variable scope

### ISSUE THREAD

**Issue by azoimide** [jimhester/lintr#330]
Linting this code should yield a warning since the function uses a variable that is neither a parameter nor local:

```
a <- 1
f <- function() {
    return(a + 1)
}

```

**Comment by Hong-Revo** [comment]
This is debatable, given that lexical scope is a feature, not a bug in R:

```r
function(lst)
{
    y <- 42
    lapply(lst, function(x) x + y
}
```

Here, the inner function will find y in the scope of the outer function without requiring it as an explicit argument.

**Comment by jimhester** [comment]
Agreed, this is used a great deal in R code, I don't think we should have a linter for this.
